aws/transit_gateway_test.go

```diff
@@ -0,0 +1,294 @@
+		},
+	})
+	require.NoError(t, err)
+
+	err = nukeAllTransitGatewayRouteTables(session, []*string{tgwRouteTable.TransitGatewayRouteTableId})
+	require.NoError(t, err)
+
+	ids, err := getAllTransitGatewayRouteTables(session, region, time.Now().Add(1*time.Hour))
+	require.NoError(t, err)
+	assert.NotContains(t, awsgo.StringValueSlice(ids), awsgo.StringValue(tgwRouteTable.TransitGatewayRouteTableId))
+}
+
+func createTestTransitGatewayVpcAttachment(t *testing.T, session *session.Session, name string) ec2.TransitGatewayVpcAttachment {
+	svc := ec2.New(session)
+
+	transitGateway := createTestTransitGateway(t, session, name)
+
+	input := &ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   awsgo.String("isDefault"),
+				Values: []*string{awsgo.String("true")},
+			},
+		},
+	}
+
+	vpcs, _ := svc.DescribeVpcs(input)
+	vpc := vpcs.Vpcs[0]
+
+	subnets := getVpcSubnets(t, session, awsgo.StringValue(vpc.VpcId))
+
+	tgwVpctAttachmentName := ec2.TagSpecification{
+		ResourceType: awsgo.String(ec2.ResourceTypeTransitGatewayAttachment),
+		Tags: []*ec2.Tag{
+			{
+				Key:   aws.String("Name"),
+				Value: aws.String(name),
+			},
+		},
+	}
+
+	param := &ec2.CreateTransitGatewayVpcAttachmentInput{
+		TagSpecifications: []*ec2.TagSpecification{&tgwVpctAttachmentName},
+		TransitGatewayId:  transitGateway.TransitGatewayId,
+		VpcId:             vpc.VpcId,
+		SubnetIds:         awsgo.StringSlice(subnets),
+	}
+
+	result, err := svc.CreateTransitGatewayVpcAttachment(param)
+	require.NoError(t, err)
+	require.True(t, len(aws.StringValue(result.TransitGatewayVpcAttachment.TransitGatewayAttachmentId)) > 0, "Could not create test Transitgateway Attachment")
+
+	//TransitGateway takes some time to be available and there isn't Waiters available yet
+	//To avoid test errors, I'm introducing a sleep call
+	time.Sleep(180 * time.Second)
+	tgwAttachment := *result.TransitGatewayVpcAttachment
+	return tgwAttachment
+}
+
+func TestGetAllTransitGatewayVpcAttachment(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	session, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region)},
+	)
+	require.NoError(t, err)
+
+	tgwName := "clud-nuke-test-" + util.UniqueID()
+	tgwAttachment := createTestTransitGatewayVpcAttachment(t, session, tgwName)
+
+	ids, err := getAllTransitGatewayVpcAttachments(session, region, time.Now().Add(1*time.Hour*-1))
+	require.NoError(t, err)
+	assert.NotContains(t, awsgo.StringValueSlice(ids), awsgo.StringValue(tgwAttachment.TransitGatewayAttachmentId))
+
+	ids, err = getAllTransitGatewayVpcAttachments(session, region, time.Now().Add(1*time.Hour))
+	require.NoError(t, err)
+	assert.Contains(t, awsgo.StringValueSlice(ids), awsgo.StringValue(tgwAttachment.TransitGatewayAttachmentId))
+
+	nukeAllTransitGatewayVpcAttachments(session, []*string{tgwAttachment.TransitGatewayAttachmentId})
+	nukeAllTransitGatewayInstances(session, []*string{tgwAttachment.TransitGatewayId})
```

Comment:
Since these return errors, we should `assert.NoError(t, err)` here.

Reply:
Implemented! Could you check again, please?